src/raw_statement.rs

```diff
@@ -143,13 +143,7 @@ impl RawStatement {
             let import_arrow2_array = arrow2::ffi::import_array_from_c(ffi_arrow2_array, arrow2_field.data_type);
 
             if let Err(err) = import_arrow2_array {
-                // When array is empty, import_array_from_c returns error with message
-                // "OutOfSpec("An ArrowArray of type X must have non-null children")
-                // Therefore, we return None when encountering this error.
-                match err {
-                    arrow2::error::Error::OutOfSpec(_) => return None,
-                    _ => panic!("Failed to import arrow2 Array from C: {}", err),
-                }
+                panic!("Failed to import arrow2 Array from C: {}", err)
```

Comment:
Not positive, but I don't believe this is valid comment anymore. It looks like `polars` did some housekeeping in the last couple of months with their `polars-arrow` package. It looks like the relevant bits [are here](https://github.com/pola-rs/polars/blob/main/crates/polars-arrow/src/ffi/array.rs#L20).

An aside I stumbled on this issue with some additional information on the changes:
https://github.com/jorgecarleitao/arrow2/pull/1476

Reply:
Sorted it out, [they're using](https://github.com/pola-rs/polars/blob/d3ecfe162e8e54fc9bb2f7062629e3d38f8a6059/crates/polars-arrow/src/ffi/array.rs#L366) `PolarsError::ComputeError` now